src/query/service/src/interpreters/interpreter_table_recluster.rs

```diff
@@ -53,26 +52,25 @@ impl Interpreter for ReclusterTableInterpreter {
         let tenant = ctx.get_tenant();
         let start = SystemTime::now();
 
+        let table = self
+            .ctx
+            .get_catalog(&plan.catalog)?
+            .get_table(tenant.as_str(), &plan.database, &plan.table)
+            .await?;
+
         // Build extras via push down scalar
-        let extras = match &plan.push_downs {
-            None => None,
-            Some(scalar) => {
-                let eb = ExpressionBuilderWithoutRenaming::create(plan.metadata.clone());
-                let pred_expr = eb.build(scalar)?;
-                Some(PushDownInfo {
-                    filters: vec![pred_expr],
-                    ..PushDownInfo::default()
-                })
-            }
+        let extras = if let Some(scalar) = &plan.push_downs {
+            let filter = scalar.as_expr()?.as_remote_expr();
+
+            Some(PushDownInfo {
+                filters: vec![filter],
+                ..PushDownInfo::default()
+            })
+        } else {
+            None
         };
 
         loop {
-            let table = self
```

Comment:
Here it is refreshing the table meta and cannot be deleted.

Reply:
thanks! fixed